src/sentences/gsa.rs

```diff
@@ -103,17 +136,33 @@ fn is_comma(x: char) -> bool {
 
 fn do_parse_empty_gsa_tail(i: &str) -> IResult<&str, GsaTail> {
     value(
-        (Vec::new(), None, None, None),
+        (Vec::new(), None, None, None, None),
         all_consuming(take_while1(is_comma)),
     )(i)
 }
 
-fn do_parse_gsa(i: &str) -> IResult<&str, GsaData> {
+fn do_parse_gsa<'a>(talker_id: &str, i: &'a str) -> IResult<&'a str, GsaData> {
     let (i, mode1) = one_of("MA")(i)?;
     let (i, _) = char(',')(i)?;
     let (i, mode2) = one_of("123")(i)?;
     let (i, _) = char(',')(i)?;
-    let (i, mut tail) = alt((do_parse_empty_gsa_tail, do_parse_gsa_tail))(i)?;
+    let (i, mut tail) = alt((
+        do_parse_empty_gsa_tail,
+        do_parse_gsa_tail_with_gnss_type,
+        do_parse_gsa_tail,
+    ))(i)?;
+
+    let gnss_type = match talker_id {
+        "GA" => GnssType::Galileo,
+        "GP" => GnssType::Gps,
+        "GL" => GnssType::Glonass,
+        "BD" | "GB" => GnssType::Beidou,
+        "GI" => GnssType::NavIC,
+        "GQ" | "PQ" | "QZ" => GnssType::Qzss,
+        "GN" => tail.4.unwrap_or(GnssType::Gps),
+        _ => tail.4.unwrap_or(GnssType::Gps),
```

Comment:
- [ ] Maybe an `Option`al field makes more sense, as this system id was introduced in NMEA 4.11?

- [ ] Also, could you please document this feature in the struct itself including the NMEA 4.11 version and how the system id is handled by us.

Not sure how it's handled by receivers in previous versions and multi-constelation view.
Here's what I found (https://gpsd.gitlab.io/gpsd/NMEA.html#_satellite_ids):

GLONASS satellite numbers come in two flavors. If a sentence has a GL talker ID, expect the skyviews to be GLONASS-only and in the range 1-32; you must add 64 to get a globally-unique NMEA ID. If the sentence has a GN talker ID, the device emits a multi-constellation skyview with GLONASS IDs already in the 65-96 range.

Reply:
If I understand the issue correctly, I'd suggest to use a `None` in the case of `GN` with no System id/gnss type.
Seems the safest approach instead of providing false information to the user or their hardware.
E.g. if they know their hardware only outputs X or Y talkers but doesn't support NMEA 4.11 system id, they will see Gps which is incorrect.